src/shared/logs/folding.py

```diff
@@ -0,0 +1,162 @@
+from collections.abc import Iterator
+from datetime import datetime
+from typing import Literal
+
+from pydantic import BaseModel
+
+from shared.logs.collections import EventCollection
+from shared.logs.raw_events import (
+    Maintainer,
+    RawMaintainerEvent,
+    RawPackageEvent,
+    RawStatusEvent,
+)
+
+
+class FoldedEvent(BaseModel):
```

Comment:
This is indistinguishable from a `RawEvent`.

I may be falling victim to premature abstraction myself with the following thought, but: it seems more ergonomic to me to derive folded events and annihilating events from the same base classes that pertain to the object of interest (suggestion, package, maintainer). Arguably these will be tiny classes, and arguably they still may change independently in the future, so I don't know. But the overlap in semantics makes it compelling to coalesce them along *that* dimension rather than anything else.

Reply:
Regarding the first remark: `RawEvent` and `FoldedEvent` are indistinguishable in content if you isolate them and compare them, but not in meaning nor usage. Having a collection of `RawEvent` and having a collection of `FoldedEvent` is meant to convey that they are to be treated differently, unlike in the design before where singular events and folded events shared the same classes, leading to confusion in the model. IMO making `FoldedEvent` inherit from `Event` would not make sense. `FoldedEvent` is to be seen as the type of what's sent to the view template. Maybe `ViewEvent`/`BatchedEvents` would be better.

I'm not sure I understand the second paragraph correctly.